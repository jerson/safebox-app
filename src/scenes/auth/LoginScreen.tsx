import React, { Fragment, useEffect } from 'react';
import { SafeAreaView, StyleSheet, StatusBar } from 'react-native';

const styles = StyleSheet.create({});

const LoginScreen: React.FC = () => {
  return (
    <Fragment>
      <StatusBar barStyle="dark-content" />
      <SafeAreaView />
    </Fragment>
  );
};

export default LoginScreen;
