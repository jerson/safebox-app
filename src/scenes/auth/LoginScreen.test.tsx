import React from 'react';
import * as renderer from 'react-test-renderer';
import LoginScreen from './LoginScreen';

const props: LoginScreen = {};
test('renders without crashing', () => {
  const snapshot = renderer.create(<LoginScreen {...props} />).toJSON();
  expect(snapshot).toBeTruthy();
});

test('render just views', () => {
  const snapshot = renderer.create(<LoginScreen {...props} />).toJSON();
  expect(snapshot).toMatchSnapshot();
});
