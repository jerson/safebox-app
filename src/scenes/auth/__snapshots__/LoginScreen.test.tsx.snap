// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render just views 1`] = `
<_class
  style={
    Array [
      Object {
        "backgroundColor": "#fafdff",
        "flex": 1,
      },
      Object {
        "backgroundColor": "#4531cd",
      },
    ]
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "minHeight": 1314,
      }
    }
    keyboardShouldPersistTaps="handled"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <_class>
      <_class
        onLayout={[Function]}
        pointerEvents="box-none"
        style={
          Object {
            "flex": 1,
            "paddingBottom": 0,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 20,
          }
        }
      >
        <_class
          style={
            Array [
              Object {
                "paddingHorizontal": 20,
              },
              Object {
                "justifyContent": "center",
              },
              Object {
                "flex": 1,
                "paddingTop": 20,
              },
              undefined,
            ]
          }
        >
          <_class
            onLayout={[Function]}
            style={
              Object {
                "paddingBottom": 0,
              }
            }
          >
            <_class
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "justifyContent": "center",
                    "overflow": "visible",
                  },
                  false,
                  Object {
                    "marginBottom": 20,
                    "marginTop": 20,
                  },
                ]
              }
            >
              <_class
                style={
                  Object {
                    "alignSelf": "center",
                    "backgroundColor": "#38414e",
                    "borderColor": "#c2d0e7",
                    "borderRadius": 22,
                    "borderWidth": 1,
                    "overflow": "hidden",
                  }
                }
              >
                <Image
                  resizeMode="stretch"
                  source={
                    Object {
                      "testUri": "../../../src/assets/images/logo.png",
                    }
                  }
                  style={
                    Object {
                      "height": 90,
                      "width": 90,
                    }
                  }
                />
              </_class>
              <Text
                style={
                  Array [
                    Object {
                      "fontFamily": "Nunito",
                      "fontSize": undefined,
                      "fontStyle": "normal",
                      "fontWeight": "400",
                    },
                    Object {},
                    Array [
                      Object {
                        "color": "#FFFFFF",
                        "fontSize": 35,
                      },
                      undefined,
                    ],
                  ]
                }
              >
                Safe
                <Text
                  style={
                    Array [
                      Object {
                        "fontFamily": "Nunito",
                        "fontSize": undefined,
                        "fontStyle": "normal",
                        "fontWeight": "700",
                      },
                      Object {},
                      undefined,
                    ]
                  }
                >
                  Box
                </Text>
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "fontFamily": "Nunito",
                      "fontSize": undefined,
                      "fontStyle": "normal",
                      "fontWeight": "300",
                    },
                    Object {},
                    Array [
                      Object {
                        "color": "#f2f4f9",
                        "fontSize": 14,
                      },
                      undefined,
                    ],
                  ]
                }
              >
                secrets always safe
              </Text>
            </_class>
            <_class
              style={
                Object {
                  "alignSelf": "center",
                  "marginBottom": 20,
                  "width": 280,
                }
              }
            >
              <_class
                style={
                  Array [
                    Object {
                      "flexDirection": "column",
                      "marginBottom": 20,
                      "marginTop": 10,
                      "position": "relative",
                    },
                    Object {
                      "marginBottom": 0,
                    },
                  ]
                }
              >
                <_class
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "paddingBottom": 4,
                      },
                      undefined,
                    ]
                  }
                />
                <_class
                  style={
                    Array [
                      Object {
                        "position": "relative",
                      },
                      undefined,
                    ]
                  }
                >
                  <TextInput
                    allowFontScaling={true}
                    autoCapitalize="none"
                    autoCompleteType="username"
                    autoCorrect={false}
                    editable={true}
                    keyboardType="default"
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    onFocus={[Function]}
                    onSubmitEditing={[Function]}
                    placeholder="Username"
                    placeholderTextColor="#65748b"
                    rejectResponderTermination={true}
                    returnKeyType="next"
                    selectionColor="#65748b"
                    style={
                      Array [
                        Object {
                          "backgroundColor": "#FFFFFF",
                          "borderColor": "#f2f4f9",
                          "borderRadius": 10,
                          "borderWidth": 1,
                          "color": "#38414e",
                          "fontFamily": "Nunito",
                          "fontSize": 15,
                          "fontStyle": "normal",
                          "fontWeight": "400",
                          "paddingBottom": 10,
                          "paddingLeft": 10,
                          "paddingRight": 8,
                          "paddingTop": 10,
                          "textDecorationLine": "none",
                        },
                        Object {
                          "paddingLeft": 40,
                        },
                        false,
                        false,
                        Object {
                          "paddingRight": 40,
                        },
                        Object {
                          "borderColor": "#4531cd",
                          "borderWidth": 1,
                        },
                      ]
                    }
                    underlineColorAndroid="transparent"
                    value=""
                  />
                  <_class
                    style={
                      Object {
                        "bottom": 1,
                        "justifyContent": "center",
                        "left": 12,
                        "position": "absolute",
                        "top": 0,
                      }
                    }
                  >
                    <Text
                      allowFontScaling={false}
                      style={
                        Array [
                          Object {
                            "color": undefined,
                            "fontSize": 12,
                          },
                          Array [
                            Object {
                              "color": "#65748b",
                              "fontSize": 20,
                            },
                            false,
                            false,
                            undefined,
                          ],
                          Object {
                            "fontFamily": "Feather",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                    >
                      ï‡¶
                    </Text>
                  </_class>
                  <_class
                    style={
                      Array [
                        Object {
                          "position": "absolute",
                          "right": -60,
                        },
                        undefined,
                      ]
                    }
                  />
                </_class>
              </_class>
              <_class
                style={
                  Array [
                    Object {
                      "flexDirection": "column",
                      "marginBottom": 20,
                      "marginTop": 10,
                      "position": "relative",
                    },
                    Object {
                      "marginBottom": 0,
                    },
                  ]
                }
              >
                <_class
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "paddingBottom": 4,
                      },
                      undefined,
                    ]
                  }
                />
                <_class
                  style={
                    Array [
                      Object {
                        "position": "relative",
                      },
                      undefined,
                    ]
                  }
                >
                  <TextInput
                    allowFontScaling={true}
                    autoCapitalize="none"
                    autoCompleteType="password"
                    autoCorrect={false}
                    blurOnSubmit={true}
                    editable={true}
                    keyboardType="default"
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    onFocus={[Function]}
                    onSubmitEditing={[Function]}
                    placeholder="Password"
                    placeholderTextColor="#65748b"
                    rejectResponderTermination={true}
                    returnKeyType="done"
                    secureTextEntry={true}
                    selectionColor="#65748b"
                    style={
                      Array [
                        Object {
                          "backgroundColor": "#FFFFFF",
                          "borderColor": "#f2f4f9",
                          "borderRadius": 10,
                          "borderWidth": 1,
                          "color": "#38414e",
                          "fontFamily": "Nunito",
                          "fontSize": 15,
                          "fontStyle": "normal",
                          "fontWeight": "400",
                          "paddingBottom": 10,
                          "paddingLeft": 10,
                          "paddingRight": 8,
                          "paddingTop": 10,
                          "textDecorationLine": "none",
                        },
                        Object {
                          "paddingLeft": 40,
                        },
                        false,
                        false,
                        Object {
                          "paddingRight": 40,
                        },
                        Object {
                          "borderColor": "#4531cd",
                          "borderWidth": 1,
                        },
                      ]
                    }
                    underlineColorAndroid="transparent"
                    value=""
                  />
                  <_class
                    style={
                      Object {
                        "bottom": 1,
                        "justifyContent": "center",
                        "left": 12,
                        "position": "absolute",
                        "top": 0,
                      }
                    }
                  >
                    <Text
                      allowFontScaling={false}
                      style={
                        Array [
                          Object {
                            "color": undefined,
                            "fontSize": 12,
                          },
                          Array [
                            Object {
                              "color": "#65748b",
                              "fontSize": 20,
                            },
                            false,
                            false,
                            undefined,
                          ],
                          Object {
                            "fontFamily": "Feather",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                    >
                      ï†‰
                    </Text>
                  </_class>
                  <_class
                    style={
                      Array [
                        Object {
                          "position": "absolute",
                          "right": -60,
                        },
                        undefined,
                      ]
                    }
                  />
                </_class>
              </_class>
              <_class
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "backgroundColor": "#9066f5",
                    "borderColor": "#9066f5",
                    "borderRadius": 10,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "margin": 0,
                    "marginTop": 20,
                    "opacity": 1,
                    "padding": 12,
                    "zIndex": 1,
                  }
                }
                testID="button_touchable"
              >
                <Text
                  style={
                    Array [
                      Object {
                        "fontFamily": "Nunito",
                        "fontSize": undefined,
                        "fontStyle": "normal",
                        "fontWeight": "400",
                      },
                      Object {},
                      Array [
                        Object {
                          "color": "#FFFFFF",
                          "fontSize": 14,
                          "textAlign": "center",
                        },
                        undefined,
                        undefined,
                      ],
                    ]
                  }
                >
                  Sign In
                </Text>
              </_class>
              <_class
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                    },
                    Object {
                      "marginTop": 20,
                    },
                  ]
                }
              >
                <_class
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#9066f5",
                        "height": 1,
                        "left": 0,
                        "opacity": 0.5,
                        "position": "absolute",
                        "right": 0,
                        "top": 12,
                      },
                      false,
                      false,
                    ]
                  }
                />
                <_class
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#4531cd",
                        "borderRadius": 10,
                        "paddingBottom": 4,
                        "paddingLeft": 8,
                        "paddingRight": 8,
                        "paddingTop": 6,
                      },
                      false,
                      false,
                    ]
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "fontFamily": "Nunito",
                          "fontSize": undefined,
                          "fontStyle": "normal",
                          "fontWeight": "400",
                        },
                        Object {},
                        Object {
                          "color": "#FFFFFF",
                          "fontSize": 13,
                          "lineHeight": 14,
                          "minWidth": 20,
                          "textAlign": "center",
                        },
                      ]
                    }
                  >
                    or
                  </Text>
                </_class>
              </_class>
              <_class
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "backgroundColor": "#6781a6",
                    "borderColor": "#6781a6",
                    "borderRadius": 10,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "margin": 0,
                    "marginTop": 20,
                    "opacity": 1,
                    "padding": 12,
                    "zIndex": 1,
                  }
                }
                testID="button_touchable"
              >
                <Text
                  style={
                    Array [
                      Object {
                        "fontFamily": "Nunito",
                        "fontSize": undefined,
                        "fontStyle": "normal",
                        "fontWeight": "400",
                      },
                      Object {},
                      Array [
                        Object {
                          "color": "#FFFFFF",
                          "fontSize": 14,
                          "textAlign": "center",
                        },
                        undefined,
                        undefined,
                      ],
                    ]
                  }
                >
                  Create account
                </Text>
              </_class>
            </_class>
          </_class>
        </_class>
      </_class>
    </_class>
  </RCTScrollView>
</_class>
`;
